scripts/test/compareBenchmarkScriptTest.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import json
+import math
+import subprocess
+
+# This script takes two JSON result files, runs the compare_benchmarks.py script, and checks that the output is as
+# expected.
+
+
+# The next two functions check for string matches instead of using math.isclose() since extracting values from
+# ANSI-colored text turned out to be too cumbersome.
+def assert_latency_equals(item_count, runtimes, latency_string):
+    avg_latency = sum(runtimes) / item_count / 1_000_000
+    assert str(round(avg_latency, 1)) in latency_string
+
+
+def assert_throughput_equals(item_count, duration, throughput_string):
+    throughput = item_count / duration * 1_000_000_000
+    assert str(round(throughput, 2)) in throughput_string
+
+
+class CompareBenchmarkScriptTest:
+    results_1 = {}
+    results_2 = {}
+    script_output = ""
+    script_output_lines = ""
+
+    def __init__(self, script_path, result_filepath_1, result_filepath_2):
+        with open(result_filepath_1) as json_file:
+            self.results_1 = json.load(json_file)
+        with open(result_filepath_2) as json_file:
+            self.results_2 = json.load(json_file)
+
+        with subprocess.Popen(
+            [script_path, result_filepath_1, result_filepath_2], stdout=subprocess.PIPE
+        ) as proc:
+            self.script_output = proc.stdout.read().decode("utf-8")
+
+        self.script_output_lines = self.script_output.splitlines()
+
+    def run(self):
+        self.check_context_table()
+        self.check_result_table()
+
+    # Checks the contents of context table which is printed above the results.
+    def check_context_table(self):
+        assert self.script_output.startswith("+Configuration Overview----")
+
+        horizontal_table_separators = [
+            i
+            for i, line in enumerate(self.script_output_lines)
+            if line.startswith("+-------")
+        ]
+        end_of_context_table = horizontal_table_separators[
+            1
+        ]  # first is end of context table header
+
+        context_1 = self.results_1["context"]
+        context_2 = self.results_2["context"]
+
+        common_keys = set(context_1.keys()) & set(context_2.keys())
+        expected_row_count = (
+            len(common_keys)
+            + (len(context_1) - len(common_keys))
+            + (len(context_2) - len(common_keys))
+        )
+        assert (
+            end_of_context_table - 3
+        ) == expected_row_count  # s -3 to substact header
```

Comment:
Looks verbose. Why not make this a one-liner?

Reply:
It's what the `black` (https://github.com/psf/black) linter does. Linting will be tackled by Jan later, so I wouldn't change any linting issues right now. Ok?

Reply:
ok 👍